File: test_cases/python/default_parameters_removed/after.py
def greet(name, greeting, punctuation):
    """Function without default parameters."""
    return f"{greeting}, {name}{punctuation}"

def calculate(x, y, z):
    """Simple calculation function."""
    return x + y * z

def process_data(data, options, debug):
    """Data processing function."""
    if debug:
        print(f"Processing {len(data)} items")
    
    return [item for item in data if item in options]

File: test_cases/javascript/default_parameters_removed/after.js
// After: Functions without default parameters
function greet(name) {
  name = name || 'Guest';
  return `Hello, ${name}!`;
}

function calculateTotal(price, quantity, tax) {
  quantity = quantity || 1;
  tax = tax || 0.1;
  return price * quantity * (1 + tax);
}

function createUser(username, email, role) {
  if (!role) {
    role = 'user';
  }
  return {
    username: username,
    email: email,
    role: role,
    createdAt: new Date()
  };
}

// Usage
const greeting = greet();
const total = calculateTotal(29.99);
const user = createUser('johndoe', 'john@example.com');
